### PR TITLE
std.zig.system: small cleanup, de-duplicate code with std.elf 

### PR DESCRIPTION
**Pull request by BratishkaErik**

### CODE REVIEW

**Comment by BratishkaErik** [comment]
Rebased on master.

**Comment by BratishkaErik** [comment]
x86_64-linux-release failure seems unrelated?

**Comment by BratishkaErik** [comment]
Rebased on master, failure is gone.

**Comment by BratishkaErik** [comment]
Rebased on master.

File: lib/std/zig/system.zig [hunk 1]
```diff
@@ -461,139 +444,127 @@ pub fn abiAndDynamicLinkerFromFile(
...
+        error.NotOpenForReading => unreachable,
+        error.SystemResources, error.Unexpected => |e| return e,
+        error.InputOutput => return error.FileSystem,
+        else => return error.UnableToReadElfFile,
+    }) |*program_header| switch (program_header.p_type) {
+        elf.PT_INTERP => {
+            got_dyn_section = true;
+            if (look_for_ld) {
+                const p_offset = program_header.p_offset;
+                const p_filesz = program_header.p_filesz;
```
Comment (andrewrk) [inline lib/std/zig/system.zig:463]: The code that you are replacing here is designed to make as few syscalls as possible. Can you share how the new code fared against the old code?

File: lib/std/zig/system.zig [hunk 2]
```diff
@@ -979,20 +927,21 @@ fn detectAbiAndDynamicLinker(
...
+    const elf_file = elf_file: {
+        // Note that shebang line can contain relative path.
         var file_name: []const u8 = "/usr/bin/env";
-        // #! (2) + 255 (max length of shebang line since Linux 5.1) + \n (1)
+        // #! (2) + 255 (max length of shebang line since Linux 5.1) + \n (1) just in case.
+        // See `man execve(2)`.
         var buffer: [258]u8 = undefined;
         while (true) {
-            const file = fs.openFileAbsolute(file_name, .{}) catch |err| switch (err) {
+            const file = cwd.openFile(file_name, .{ .mode = .read_only }) catch |err| switch (err) {
```
Comment (andrewrk) [inline lib/std/zig/system.zig:944]: `openFileAbsolute` is correct when the file path is an absolute path, as it is with `/usr/bin/env` as well as any shebang line encountered.

File: lib/std/elf.zig [hunk 3]
```diff
@@ -455,6 +455,54 @@ pub const ET = enum(u16) {
     pub const HIPROC = 0xffff;
 };
 
+/// All integers are native endian.
+pub const SectionHeader = struct {
```
Comment (andrewrk) [inline lib/std/elf.zig:459]: Opinionated code like this should not go into std.elf but somewhere else such as std.zig.elf perhaps. However, why is this even needed? It's redundant with Elf64_Shdr.

**Review by andrewrk — CHANGES_REQUESTED** [review]
It looks like there is a bug fix here buried among "cleanup" commits. It is unclear to me how the new code is better than the old code.

I have 2 requests for you:
1. Can you submit the bug fix independently?
2. Can you explain what is improved in the other commits without using the word "clean"?

**Pull request closed without merging**